layers/ntp.go

```diff
@@ -0,0 +1,450 @@
+	ReferenceID        uint32 // ID code of reference clock [0,2^32-1].
+	ReferenceTimestamp uint64 // Most recent timestamp from the reference clock.
+	OriginTimestamp    uint64 // Local time when request was sent from local host.
+	ReceiveTimestamp   uint64 // Local time (on server) that request arrived at server host.
+	TransmitTimestamp  uint64 // Local time (on server) that request departed server host.
+
+	// FIX: This package should analyse the extension fields and represent the extension fields too.
+	ExtensionBytes []byte // Just put extensions in a byte slice.
+}
+
+//******************************************************************************
+
+// LayerType() returns the layer type of the NTP object, which is LayerTypeNTP.
+//
+func (d *NTP) LayerType() gopacket.LayerType {
+	return LayerTypeNTP
+}
+
+//******************************************************************************
+
+// decodeNTP() analyses a byte slice and attempts to decode it as an NTP
+// record of a UDP packet.
+//
+// If it succeeds, it loads p with information about the packet and returns nil.
+// If it fails, it returns an error (non nil).
+//
+// This function is employed in layertypes.go to register the NTP layer.
+//
+func decodeNTP(data []byte, p gopacket.PacketBuilder) error {
+
+	// Attempt to decode the byte slice.
+	//
+	d := &NTP{}
+	err := d.DecodeFromBytes(data, p)
+	if err != nil {
+		return err
+	}
+
+	// If the decoding worked, add the layer to the packet and set it
+	// as the application layer too, if there isn't already one.
+	//
+	p.AddLayer(d)
+	p.SetApplicationLayer(d)
+
+	return nil
+}
+
+//******************************************************************************
+
+// decodeNTP() analyses a byte slice and attempts to decode it as an NTP
+// record of a UDP packet.
+//
+// Upon succeeds, it loads the NTP object with information about the packet
+// and returns nil.
+// Upon failure, it returns an error (non nil).
+//
+func (d *NTP) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+
+	// If the data block is too short to be a NTP record, then return an error.
+	//
+	if len(data) < ntp_minimum_record_size_in_bytes {
+		df.SetTruncated()
+		return fmt.Errorf("NTP packet too short")
+	}
+
+	// RFC 5905 does not appear to define a maximum NTP record length.
+	// The protocol allows "extension fields" to be included in the record,
+	// and states about these fields:"
+	//
+	//     "While the minimum field length containing required fields is
+	//      four words (16 octets), a maximum field length remains to be
+	//      established."
+	//
+	// For this reason, the packet length is not checked here for being too long.
+
+	// NTP type embeds type BaseLayer which contains two fields:
+	//    Contents is supposed to contain the bytes of the data at this level.
+	//    Payload is supposed to contain the payload of this level.
+	// Here we set the baselayer to be the bytes of the NTP record.
+	//
```

Comment:
Sorry, I'm totally okay with, for example:

```go
// blah blah blah
//
// blah blah blah
code
```

I was asking for removal of the `// <empty line>` comments between comment blocks and code, like these: 

```go
// blah blah blah
//
// blah blah blah
// <extraneous empty line, commented out>
code
```